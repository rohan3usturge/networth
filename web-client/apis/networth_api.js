import axios from "axios";

const baseUrl = process.env.API_BASE_URL;

const getDefaultLineItems = async () => {
  const response = await axios.get(`${baseUrl}/networth`, {
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
    },
  });
  return response.data;
};

const getNetWorth = async (netWorthCalculateRequest) => {
  const response = await axios.post(
    `${baseUrl}/networth`,
    netWorthCalculateRequest,
    {
      headers: {
        Accept: "application/json",
        "Content-Type": "application/json",
      },
    }
  );
  return response.data;
};

const getCurrencies = async () => {
  const response = await axios.get(`${baseUrl}/networth/currencies`, {
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
    },
  });
  return response.data;
};

export const NETWORTH_API = {
  getDefaultLineItems,
  getNetWorth,
  getCurrencies,
};
