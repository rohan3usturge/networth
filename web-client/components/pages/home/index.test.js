import Home, { getStaticProps } from "./index";
import { shallow } from "enzyme";
import { NETWORTH_API } from "../../../apis/networth_api";

jest.mock("../../../apis/networth_api");

describe("HomePage", () => {
  it("should render properly and match snapshot", () => {
    const wrapper = shallow(<Home portfolio={{}} currencies={[]} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should getStaticProps call correct apis", async () => {
    const currencies = ["CAD"];
    const lineItems = { assets: [], liabilities: [] };
    const networthResponse = {
      netWorth: {},
      lineItems,
      currencyCode: "CAD",
    };

    NETWORTH_API.getCurrencies.mockImplementation(() => currencies);
    NETWORTH_API.getDefaultLineItems.mockImplementation(() => lineItems);
    NETWORTH_API.getNetWorth.mockImplementation(() => networthResponse);

    await getStaticProps({ locale: "en-ca" });

    expect(NETWORTH_API.getCurrencies).toHaveBeenCalledTimes(1);
    expect(NETWORTH_API.getDefaultLineItems).toHaveBeenCalledTimes(1);
    expect(NETWORTH_API.getNetWorth).toHaveBeenCalledTimes(1);
  });
});
