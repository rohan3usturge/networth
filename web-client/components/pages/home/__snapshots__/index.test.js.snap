// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage should render properly and match snapshot 1`] = `
<section
  className="bg-light"
>
  <Header
    currencies={Array []}
    portfolio={Object {}}
  />
  <div
    className="mt-5"
  />
  <main
    className="container-md pl-1 pr-1"
  >
    <div>
      <Home
        currencies={Array []}
        portfolio={Object {}}
      />
    </div>
  </main>
  <Footer />
</section>
`;
